diff --git a/src/lib/token.ts b/src/lib/token.ts
index 6a683a6..0d88848 100644
--- a/src/lib/token.ts
+++ b/src/lib/token.ts
@@ -1,3 +1,5 @@
+import { generateKeyPairSigner } from '@solana/kit';
+import { getTransferSolInstruction } from '@solana-program/system';
 import { Connection, Keypair, PublicKey, sendAndConfirmTransaction, Signer, SystemProgram, Transaction, TransactionInstruction, TransactionMessage, VersionedTransaction } from "@solana/web3.js";
 import { makeKeypairs } from "./keypair";
 import { createAssociatedTokenAccountIdempotentInstruction, createInitializeInstruction, createInitializeMetadataPointerInstruction, createInitializeMint2Instruction, createInitializeMintInstruction, createMintToInstruction, ExtensionType, getAssociatedTokenAddressSync, getMinimumBalanceForRentExemptMint, getMintLen, LENGTH_SIZE, MINT_SIZE, TOKEN_2022_PROGRAM_ID, TOKEN_PROGRAM_ID, TYPE_SIZE } from "@solana/spl-token";
@@ -43,11 +45,7 @@ export const createAccountsMintsAndTokenAccounts = async (
   });
 
   const sendSolInstructions: Array<TransactionInstruction> = users.map((user) =>
-    SystemProgram.transfer({
-      fromPubkey: payer.publicKey,
-      toPubkey: user.publicKey,
-      lamports,
-    }),
+    getTransferSolInstruction({ source: payer.publicKey, destination: user.publicKey, amount: lamports }),
   );
 
   // Airdrops to user
@@ -115,7 +113,7 @@ export const makeTokenMint = async (
   updateAuthority: PublicKey = mintAuthority.publicKey,
   freezeAuthority: PublicKey | null = null,
 ) => {
-  const mint = Keypair.generate();
+  const mint = await generateKeyPairSigner();
 
   if (!Array.isArray(additionalMetadata)) {
     additionalMetadata = Object.entries(additionalMetadata);
